syntax = 'proto3';

package proto;

message Message {

    // sender client
    string id = 1;

    // inhoud vant berricht
    string content = 2;

    // peer_user is the receiver of the message.
    string peer_user = 4;
}


message Connect {

    // als wie je connect (client)
    string id = 1;
}

message Close{}

service Broadcast {
    rpc CreateStream(Connect) returns (stream Message);
    rpc BroadcastMessage(Message) returns (Close);
}